### PR TITLE
docs: document unit for --with-gas-price

### PR DESCRIPTION
**Pull request by PaulRBerg**
## Motivation

It is not clear in which unit the gas price should be passed when running `forge script --with-gas-price`.

## Solution

Mention the unit at the end of the description - "in wei".

### CODE REVIEW

**Comment by DaniPopes** [comment]
That's the default yes, you can also specify the unit after the value like "1 gwei"; see `value` below

**Review by Evalir — APPROVED** [review]
should be g2g. will leave up to @DaniPopes & rest of team to approve and merge this :)

File: crates/cli/src/opts/transaction.rs [hunk 1]
```diff
@@ -10,7 +10,12 @@ pub struct TransactionOpts {
     #[arg(long, env = "ETH_GAS_LIMIT")]
     pub gas_limit: Option<U256>,
 
-    /// Gas price for legacy transactions, or max fee per gas for EIP1559 transactions.
+    /// Gas price for legacy transactions, or max fee per gas for EIP1559 transactions, either
+    /// specified in wei, or as a string with a unit type.
+    ///
+    ///
+    ///
```
Comment (DaniPopes) [inline crates/cli/src/opts/transaction.rs:17]:
```suggestion
```

File: crates/script/src/lib.rs [hunk 2]
```diff
@@ -169,7 +169,12 @@ pub struct ScriptArgs {
     #[arg(long)]
     pub json: bool,
 
-    /// Gas price for legacy transactions, or max fee per gas for EIP1559 transactions.
+    /// Gas price for legacy transactions, or max fee per gas for EIP1559 transactions, either
+    /// specified in wei, or as a string with a unit type.
+    ///
+    ///
+    ///
```
Comment (DaniPopes) [inline crates/script/src/lib.rs:176]:
```suggestion
```

**Review by DaniPopes — APPROVED** [review]

**Review by mattsse — APPROVED** [review]

**Pull request merged**